tests/jobs/pyscal/test_pyscal.py

```diff
@@ -226,6 +226,34 @@ def test_fm_pyscal_argparse(tmpdir, monkeypatch):
                 3,
             ],
         ),
+        (
+            # Wrong usage of brackets in parameter names
```

Comment:
Changes look good, but would it make sense to split this into two tests? Seems it is testing two different things, one is errors thrown in this repo, the other is borderline testing `pyscal`, dont mind doing that, but splitting it in two tests does make it easier to spot where the error is coming from. Would also be great to use `with pytest.raises(raises), match=expected_error:` on the errors originating from this end, and using `caplog` for the integration tests against `pyscal`

Reply:
For the test-splitting, are you then thinking of `ValueError` vs `SystemExit` ? I did not get what to split in the commit diff alone.

Reply:
I misread the test slightly, what I meant was something like this where we assert on the contents of the log:
```
@pytest.mark.parametrize(
    "msg, raises, dframe, runargs",
    [
        (
            "Writing to a file we don't have permission to write to",
            (OSError, IOError),
            EXAMPLE_STATIC_DFRAME,
            [
                "relperm-input.csv",
                "/relperm.inc",
                "",
                "__NONE__",
                "__NONE__",
                "sgof",
                1,
            ],
        ),
        (
            # Reading a file that does not exist
            "not-existing.csv does not exist",
            (ValueError, SystemExit),
            None,
            ["not-existing.csv", "relperm.inc", "", "__NONE__", "__NONE__", "sgof", 1],
        ),
        (
            # Specifying a sheet that does not exist
            "Non-existing sheet-name foo_sheet provided",
            (Exception, SystemExit),
            # Pyscal 0.4.0 raises an UnboundLocalError in this case, but with correct
            # error messages upfront. The exact exception might improve, so just assert
            # anything is thrown:
            EXAMPLE_STATIC_DFRAME,
            ["file.xlsx", "rel.inc", "foo_sheet", "__NONE__", "__NONE__", "sgof", 1],
        ),
        (
            # GasOil-interpolation without WaterOil, this is not supported
            "WaterOil interpolation parameter missing",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "__NONE__",
                "INTERPOLATE_GO",
                "sgof",
                1,
            ],
        ),
        (
            # Parameter name not found
            "FOOOO not found in parameters",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "FOOOO",
                "INTERPOLATE_GO",
                "sgof",
                1,
            ],
        ),
        (
            # Parameter name not found
            "INTERPOLATE_GOOOOOO not found",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "INTERPOLATE_WO",
                "INTERPOLATE_GOOOOOO",
                "sgof",
                1,
            ],
        ),
        (
            # Wrong magic name for interpolation parameter
            "__OPTIM__ not found in parameters",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "__OPTIM__",
                "__NONE__",
                "sgof",
                1,
            ],
        ),
        (
            # Wrong sgof type
            "Only supports sgof or slgof",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "INTERPOLATE_WO",
                "INTERPOLATE_GO",
                "sgofffff",
                1,
            ],
        ),
        (
            # Wrong family type
            "Family must be either 1 or 2",
            (ValueError, SystemExit),
            EXAMPLE_SCAL,
            [
                "scal-input.csv",
                "relperm.inc",
                None,
                "INTERPOLATE_WO",
                "INTERPOLATE_GO",
                "sgof",
                3,
            ],
        ),
    ],
)
def test_fm_pyscal_errors(msg, raises, dframe, runargs, tmpdir, caplog):
    """Test that fm_pyscal or pyscal gives correct errors to
    common misusages"""
    tmpdir.chdir()
    if dframe is not None:
        if runargs[0].endswith("csv"):
            dframe.to_csv(runargs[0])
        else:
            dframe.to_excel(runargs[0])
    if dframe is not None and "CASE" in dframe:
        with open("parameters.txt", "w") as p_file:
            p_file.write("INTERPOLATE_WO 0.1\nINTERPOLATE_GO 0.5")
    with pytest.raises(raises):
        run(*runargs)
    with caplog.at_level("ERROR"):
        assert msg in caplog.text
```